### PR TITLE
FEAT: Handle & display mob

### PR DESCRIPTION
**Pull request by theophile-jr**
Feature the mob, he throws some bullets toward the player, and moves on the y axis. He appears after a few seconds on the screen. 
image

### CODE REVIEW

File: Engine/Client/Src/Systems/DrawTextureRect/DrawTextureRectSystem.cpp [hunk 1]
```diff
@@ -38,12 +40,19 @@ void DrawTextureRectSystem::_drawTextureRect(ECS::Registry& reg, int idxPacketEn
         std::shared_ptr<ScaleComponent> scale = (scaleComponents.size() >= entity + 1) ?
             std::dynamic_pointer_cast<ScaleComponent>(scaleComponents[entity]) : nullptr;
 
+        std::shared_ptr<ColourComponent> colour = (colourComponents.size() >= entity + 1) ?
+            std::dynamic_pointer_cast<ColourComponent>(colourComponents[entity]) : nullptr;
+
+        // Color color = {255, 255, 255, 255};
```
Comment (mathieurobert1) [inline Engine/Client/Src/Systems/DrawTextureRect/DrawTextureRectSystem.cpp:46]: Remove this useless comment

File: Engine/Shared/Interface/IGraphic.hpp [hunk 2]
```diff
@@ -125,7 +129,7 @@ class IGraphic {
          * @param a Color value a.
          */
         virtual void drawText(std::string text, float posx, float posy, int fontSize, std::string fontPath,
-            unsigned char r = 0, unsigned char g = 0, unsigned char b = 0, unsigned char a = 0) = 0;
+            unsigned char r = 255, unsigned char g = 255, unsigned char b = 255, unsigned char a = 255) = 0;
```
Comment (mathieurobert1) [inline Engine/Shared/Interface/IGraphic.hpp:132]: For the text, please let the default black colour.

File: GraphicLibrary/Src/GraphicLib.hpp [hunk 3]
```diff
@@ -147,7 +151,7 @@ class GraphicLib : public IGraphic {
          * @param a Color value a.
          */
         void drawText(std::string text, float posx, float posy, int fontSize, std::string fontPath,
-            unsigned char r = 0, unsigned char g = 0, unsigned char b = 0, unsigned char a = 0);
+            unsigned char r = 255, unsigned char g = 255, unsigned char b = 255, unsigned char a = 255);
```
Comment (mathieurobert1) [inline GraphicLibrary/Src/GraphicLib.hpp:154]: For the text, please let the black colour by default.

File: R-TypeGame/Systems/Mob/MobInitSystem.cpp [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    if (position2D) {
+        reg.register_component<IComponent>(position2D->getType());
+        reg.set_component<IComponent>(idxPacketEntities, position2D, position2D->getType());
+    }
+
+    // std::shared_ptr<SpeedComponent> speed = parseSpeed(PATH_JSON);
+    // if (speed) {
+    //     reg.register_component<IComponent>(speed->getType());
+    //     reg.set_component<IComponent>(idxPacketEntities, speed, speed->getType());
+    // }
```
Comment (mathieurobert1) [inline R-TypeGame/Systems/Mob/MobInitSystem.cpp:58]: Please remove this useless comment.

**Review by mathieurobert1 — CHANGES_REQUESTED** [review]
Some changes.

File: R-TypeGame/Systems/Mob/MobInitSystem.cpp (hunk 4 above)
Comment (Peralban) [inline R-TypeGame/Systems/Mob/MobInitSystem.cpp:58]: You can delete that

**Review by Peralban — CHANGES_REQUESTED** [review]
Just one RC. But well done. That is a great work.

**Review by Peralban — APPROVED** [review]

**Comment by theophile-jr** [comment]
Removing thomas from the reviewers, he is not available and we must merge this branch.

**Pull request merged**